src/main/java/org/kohsuke/github/GHCheckRuns.java

```diff
@@ -0,0 +1,32 @@
+package org.kohsuke.github;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents the response of listing check-runs for given ref.
+ */
+public class GHCheckRuns {
```

Comment:
FYI, I don't love the behavior of `PageSearchIterable` currently. The Iterable has methods that I would rather see done elsewhere. 


This method relies on a cached result, meaning the Iterable can get out of sync with underlying search results. 😭  
https://github.com/github-api/github-api/blob/bccae94c7a740fa7ba297fd6eb2a6782ed156a98/src/main/java/org/kohsuke/github/PagedSearchIterable.java#L48-L50

Reply:
Thanks, I didn't spot that :-) Is it OK to create a search builder class - with one method specifying the `ref` and `not` extending `GHSearchBuilder` - so that it's similar to other search builders even though it's not using the search api? Or should I not mix it with search other builders? Or any other idea? :-)

Reply:
I don't have any strong requirement here except that the api properly support paging. 

No, it shouldn't extend search builder, just use a the same basic structure.

Reply:
I created a temp commit here https://github.com/avano/github-api/commit/6ba7b3df8a75dc63d1fd20c48b202821dba0a8f9 - can you please check if I understood correctly?

Reply:
Yes, that looks right to me.